Libraries/DirectMLX.h

```diff
@@ -3470,9 +3475,43 @@ namespace dml
         return out;
     }
 
-    // 
-    // TODO: NonZeroCoordinates
-    // 
+    struct NonZeroCoordinatesOutputs
+    {
+        Expression count;
+        Expression coordinates;
+    };
+    inline NonZeroCoordinatesOutputs NonZeroCoordinates(
+        Expression input)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        TensorDesc outputCountTensor(DML_TENSOR_DATA_TYPE_UINT32, {1,1,1,1}, builder->GetTensorPolicy()); // Same as input
+
+        uint32_t effectiveRank = -1;
+        auto inputTensorSizes = inputTensor.sizes;
+        uint32_t totalElements = 1;
+        for(int i = 0; i<inputTensorSizes.size(); i++) {
```

Comment:
```suggestion
for (int i = 0; i < inputTensorSizes.size(); ++i)
```

Consistent spacing with elsewhere.

Reply:
`for(` -> `for (`
`) {` -> `)\n
{`